bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/MagentoGraphqlClient.java

```diff
@@ -0,0 +1,101 @@
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+package com.adobe.cq.commerce.core.components.internal.models.v1;
+
+import java.util.Collections;
+
+import org.apache.http.Header;
+import org.apache.http.message.BasicHeader;
+import org.apache.sling.api.resource.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.adobe.cq.commerce.graphql.client.GraphqlClient;
+import com.adobe.cq.commerce.graphql.client.GraphqlRequest;
+import com.adobe.cq.commerce.graphql.client.GraphqlResponse;
+import com.adobe.cq.commerce.graphql.client.RequestOptions;
+import com.adobe.cq.commerce.magento.graphql.Query;
+import com.adobe.cq.commerce.magento.graphql.gson.Error;
+import com.adobe.cq.commerce.magento.graphql.gson.QueryDeserializer;
+import com.day.cq.commons.inherit.ComponentInheritanceValueMap;
+import com.day.cq.commons.inherit.HierarchyNodeInheritanceValueMap;
+import com.day.cq.commons.inherit.InheritanceValueMap;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+
+/**
+ * This is a wrapper class for {@link GraphqlClient}. The constructor adapts a {@link Resource} to
+ * the GraphqlClient class and also looks for the <code>cq:magentoStore</code> property on the resource
+ * path in order to set the Magento <code>Store</code> HTTP header. This wrapper also sets the custom
+ * Magento Gson deserializer from {@link QueryDeserializer}.
+ */
+public class MagentoGraphqlClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MagentoGraphqlClient.class);
+
+    public static final String STORE_CODE_PROPERTY = "cq:magentoStore";
+
+    private GraphqlClient graphqlClient;
+    private RequestOptions requestOptions;
+
+    /**
+     * Instantiates and returns a new MagentoGraphqlClient.
+     * This method returns <code>null</code> if the client cannot be instantiated.
+     * 
+     * @param resource The JCR resource to use to adapt to the lower-level {@link GraphqlClient}.
+     * @return A new MagentoGraphqlClient instance.
+     */
+    public static MagentoGraphqlClient create(Resource resource) {
+        try {
+            return new MagentoGraphqlClient(resource);
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
```

Comment:
I don't think this should be an error. It may look like an error for the create method but not for an adaptTo. I prefer logging a warning on the call site if I don't get what I expect or I don't log anything if I decide so in any case I would live it up to the client to handle null as they want. 
By returning null I did my job, and that's also in the JavaDoc. There's no need to log anything here.

Reply:
ok I disagree, how do we resolve that? Did you decide to sabotage this PR?